search/services/index/advanced.py

```diff
@@ -0,0 +1,141 @@
+"""Supports the advanced search feature."""
+
+from typing import Any
+
+from elasticsearch_dsl import Search, Q, SF
+from elasticsearch_dsl.query import Range, Match, Bool
+
+from search.domain import AdvancedQuery, Classification
+
+from .prepare import SEARCH_FIELDS
+from .util import sort
+
+
+def advanced_search(search: Search, query: AdvancedQuery) -> Search:
```

Comment:
Not necessarily germane, but hitting the back button from advanced search results on Chrome Mac OS X and getting no results does not seem to repopulate search conditions added with the add additional field + button.  This may be intended or low impact, should I file an issue?
https://beta.arxiv.org/search/advanced?advanced=1&terms-0-operator=AND&terms-0-term=elec*&terms-0-field=title&terms-1-operator=AND&terms-1-term=fourier+AND+transform&terms-1-field=title&classification-eess=y&classification-physics_archives=all&date-filter_by=specific_year&date-year=2017&date-from_date=&date-to_date=&size=50&order=

Reply:
This is intended, in the sense that preserving input on browser back was not considered a requirement, so this is the expected behavior. The intended route for revising the query was for the user to click on the link beside "refine your query"